tests/native_tests/defclass.hy

```diff
@@ -109,3 +109,17 @@
   (assert (= a.y 2))
   (assert foo 2)
   (assert (.greet a) "hello"))
+
+(defn test-defclass-implicit-nil-for-init []
+  "NATIVE: test that defclass adds an implicit nil to --init--"
+  (defclass A []
+    [--init-- (fn [self] (setv self.x 1) 42)])
+  (defclass B []
+    (defn --init-- [self]
+      (setv self.x 2)
+      42))
```

Comment:
I'd also add a test for the actual return value of `__init__`, but it's not really important (and writing such a test in Hy might be dangerous for your eyes :))

For example, in Python:

```py
>>> repr(A.__init__(A()))
'None'
```